L1Trigger/DemonstratorTools/src/codecs_vertices.cc

```diff
@@ -0,0 +1,21 @@
+
+#include "L1Trigger/DemonstratorTools/interface/codecs/vertices.h"
+
+namespace l1t::demo::codecs {
+
+  std::vector<l1t::Vertex> decodeVertices(const l1t::demo::BoardData::Channel& frames) {
+    std::vector<l1t::Vertex> vertices;
+
+    for (const auto& x : frames) {
+      if (not(x.data & 1))
+        break;
+
+      // TODO: Replace with complete implementation
+      Vertex vertex(0.0, ap_fixed<15, 6>(ap_uint<64>(x.data)(15, 1)), {});
```

Comment:
Shouldn't the sizes for the `ap_fixed<15,6>` and `ap_uint<64>` types be defined somewhere as a names constant? Perhaps the VertexWord data format.

Reply:
Yes. From looking at `VertexWord.h`, I guess it would be best to use `VertexWord::vtxz0_t` and `VertexWord::vtxword_t` respectively; does that match how you expected those typedefs to be used?

Reply:
Yes, that's right.

Reply:
I've updated this function to extract all fields defined in the `VertexWord` class